tests/python/gaia-ui-tests/gaiatest/apps/base.py

```diff
@@ -148,7 +151,15 @@ def tap_element_from_system_app(self, element=None, add_statusbar_height=False):
     def keyboard(self):
         from gaiatest.apps.keyboard.app import Keyboard
         return Keyboard(self.marionette)
+    
+    @property
+    def name(self):
+        raise NotImplementedError("{} should implement the class attribute called 'name'".format(self.__class__)  )
 
+    @property
+    def manifest_url(self):
+        app_name_to_use = self.manifest_url_app if self.manifest_url_app else self.name.lower()
```

Comment:
From what I see [there](https://github.com/mozilla-b2g/gaia/pull/31259/files#diff-5335dc659920c9431033755d5bf5f557R15), `manifest_url_app` should be used this way.

At the moment, with the contact app, we end up with that `manifest_url`:
``` python
'app://'communications.gaiamobile.org/contacts/index.htmlgaiamobile.org/manifest.webapp'
```
That doesn't look correct to me.

For a better simplicity, I'd just have
``` python
@property
def manifest_url(self):
    return '{}{}.{}/manifest.webapp'.format(self.DEFAULT_PROTOCOL, self.name.lower(), self.DEFAULT_APP_HOSTNAME)
```

And leave the definition in Contacts, this way:
``` python
manifest_url = '{}communications.{}/contacts/index.html'.format(self.DEFAULT_PROTOCOL, self.DEFAULT_APP_HOSTNAME)
```